planet/src/main/webapp/places.js

```diff
@@ -22,106 +29,133 @@ function initMap() {
+            });
+    } else {
+        alert('Error: Your browser doesn\'t support geolocation.'); 
+    }
 }
 
-/** Create new markers and display new search */
+/** Update markers and places when new search is performed */
 function updateSearch() {    
-    let input = document.getElementById('pac-input');
-    let searchBox = new google.maps.places.SearchBox(input);
-
-    // Set position of the search bar onto the map
-    map.controls[google.maps.ControlPosition.TOP_LEFT].push(input);
-
     // Bias the SearchBox results towards current map's viewport.
     map.addListener('bounds_changed', function() {
         searchBox.setBounds(map.getBounds());
     });
 
-    let markers = [];
-    let placeNames = [];
-
     // Listener for when user selects new location
     searchBox.addListener('places_changed', function() {
-    let places = searchBox.getPlaces();
+        let places = searchBox.getPlaces();
         if (places.length == 0) {
             return;
         }
         // Delete old markers
         markers.forEach(function(marker) {
             marker.setMap(null);
         });
+        // Clear array of old markers and place information
         markers = [];
-        placeNames = [];
-
-        // For each place, get the icon, name and location.
-        let bounds = new google.maps.LatLngBounds();
-        places.forEach(function(place) {
-            if (!place.geometry) {
-              console.log("Returned place contains no geometry");
-              return;
-            }
-            // Add search results to a list
-            placeNames.push(place.name + "\n"); 
-
-            let icon = {
-              url: place.icon,
-              size: new google.maps.Size(71, 71),
-              origin: new google.maps.Point(0, 0),
-              anchor: new google.maps.Point(17, 34),
-              scaledSize: new google.maps.Size(25, 25)
-            };
+        placeInfo = [];
 
-            // Create a marker for each place.
-            markers.push(new google.maps.Marker({
-              map: map,
-              icon: icon,
-              title: place.name,
-              position: place.geometry.location
-            }));
-
-            if (place.geometry.viewport) {
-              bounds.union(place.geometry.viewport);
-            } else {
-              bounds.extend(place.geometry.location);
-            }
-        });
+        bounds = new google.maps.LatLngBounds();
+        placeDetails(places);
         map.fitBounds(bounds);
-        listResults(placeNames);
+        listResults(placeInfo);
     });
 }
 
-function handleLocationError(browserHasGeolocation, infoWindow, pos) {
-    infoWindow.setPosition(pos);
-    infoWindow.setContent(browserHasGeolocation ?
-            'Error: The Geolocation service failed.' :
-            'Error: Your browser doesn\'t support geolocation.');
-    infoWindow.open(map);
+/** Create markers and add details to each place */
+function placeDetails(places) {
```

Comment:
nit: rephrase this into a verb or verb phrase: https://google.github.io/styleguide/jsguide.html#naming-method-names

e.g. addPlaceDetails or populatePlaceDetails